lib/overlay.nix

```
@@ -0,0 +1,25 @@
+final: prev:
```

```suggestion
self: final: prev:
```

In the form of a proto-overlay, I beleive this would obsolete the need for a contrived indirection through the registry.

Exposed as `self.replOverlay` to avoid overlay checks.